network/main.tf

```diff
@@ -0,0 +1,126 @@
+# --- network/main.tf ---
+
+data "aws_availability_zones" "available" {}
+
+resource "random_integer" "random" {
+    min = 1
+    max = 10
+
+}
+resource "random_shuffle" "az_list" {
+    input = data.aws_availability_zones.available.names
+    result_count = var.max_subnets
+}
+resource "aws_vpc" "atul_vpc" {
+    cidr_block = var.vpc_cidr
+    enable_dns_hostnames = true 
+    enable_dns_support = true
+
+    tags = {
+        Name = "atul_vpc-${random_integer.random.id}"
+    }
+}
+
+resource "aws_subnet" "atul_public_subnet" {
+    count = var.public_sn_count
+    vpc_id = aws_vpc.atul_vpc.id
+    cidr_block = var.public_cidrs[count.index]
+    map_public_ip_on_launch = true
+    availability_zone = random_shuffle.az_list.result[count.index]
+
+
+    tags = {
+        Name = "atul-public_${count.index + 1}"
+    }
+}
+
+resource "aws_route_table_association" "atul_public_association" {
+
+    count = var.public_sn_count
+    subnet_id = aws_subnet.atul_public_subnet.*.id[count.index]
+    route_table_id = aws_route_table.atul_public_rt.id
+  
+}
+
+
+resource "aws_subnet" "atul_private_subnet" {
+    count =var.private_sn_count
+    vpc_id = aws_vpc.atul_vpc.id
+    cidr_block = var.private_cidrs[count.index]
+    map_public_ip_on_launch = false
+    availability_zone = random_shuffle.az_list.result[count.index]
+
+
+
+    tags = {
+        Name = "atul-private_${count.index + 1}"
```

Comment:
The `env_code` variable should be used in all the Name tags. Otherwise, you will lose track of which resource belongs to which environment when haveing more than one deployment